gen_proc_macro/src/rc.rs

```diff
@@ -0,0 +1,128 @@
+use proc_macro::TokenStream;
+use proc_macro_error::abort_call_site;
+use syn::{
+    parse_str,
+    punctuated::Punctuated,
+    token::Comma,
+    Expr,
+    FnArg,
+    Ident,
+    Pat,
+    Stmt,
+    Type,
+};
+
+use crate::common::replace_yield_cls;
+
+/// Mutates the input `Punctuated<FnArg, Comma>` to a lifetimeless `co:
+/// Co<{type}>`.
+pub(crate) fn add_coroutine_arg(punct: &mut Punctuated<FnArg, Comma>, co_ty: String) {
+    if !punct.iter().any(|input| {
```

Comment:
can you assign this condition to a var instead of having a multiline if statement? This is a tough one to skim.